### PR TITLE
Add Reentrancy Protection And Update Tests

### PR DESCRIPTION
**Pull request by FaybianB**
Adds `norreantrant` modifier, updates tests and allows owner to withdraw balance.

### CODE REVIEW

File: contracts/AckToEarn.sol [hunk 1]
```diff
@@ -74,6 +77,7 @@ contract AckToEarn is Ownable, ReentrancyGuard {
 
         bidderBids[msg.sender][newBidId] = bid;
         recipientBids[recipient][newBidId] = bid;
+        balances[owner()] = ownerAmount;
```
Comment (pardhaponugoti) [inline contracts/AckToEarn.sol:80]: should this be `+= ownerAmount`?

File: test/AckToEarn.js [hunk 2]
```diff
@@ -166,6 +171,7 @@ describe("AckToEarn", function () {
         "Some message",
         otherAccount.address,
         "bidder@email.com",
+        "fileCid",
```
Comment (pardhaponugoti) [inline test/AckToEarn.js:174]: sorry!

**Review by pardhaponugoti — APPROVED** [review]

File: contracts/AckToEarn.sol (hunk 1 above)
Comment (FaybianB) [reply contracts/AckToEarn.sol:80]: Yes, it should, good catch.

File: test/AckToEarn.js (hunk 2 above)
Comment (FaybianB) [reply test/AckToEarn.js:174]: No problem, I realized after yours was already merged.

**Pull request merged**